### ISSUE TITLE
Source tarball consistency

### ISSUE THREAD

**Issue by dvzrv** [pyocd/cmsis-pack-manager#226]
Hi @mbrossard @flit :wave: 

I package this project for Arch Linux. We are currently rebuilding all packages against Python 3.12 and I noticed discrepancies for the sources of this project.

I have downloaded and packaged with an auto-generated source tarball on 2023-12-01 (https://gitlab.archlinux.org/archlinux/packaging/packages/python-cmsis-pack-manager/-/commit/46e660d0dee2674d74d33951988f1fa62857fdf1) with the blake2 sum `c623027d77f85957bd0328b8fbaa278d93445edd47f480da1aa7738b4047e9728411a2b8b9963ad6c3a53a85307fd935760b63fab384e7eb490f84fa9511ab5a`

I downloaded the source tarball again today, and it is now `52d04392c5a858531acd7fec2a1a1603ce5962a0e9868363eb62c2844bbf34b095cebddb4c0396eee7fd4ec68f66854a8bb94643ca124ac58db4e5e34b132f72`.

(to make sure I didn't accidentally checksum the sdist tarball from PyPI, which was used for previous versions, I downloaded and checked: its checksum is `f98fc68ec9dda53ee32b599837286f52ec57c2372eac7bc2ad049d46378b59f3fea3bb60c6de844d7d136e5fd02045f85fd51102106f2af9821c89609e50eb56`)

Usually these things happen if upstreams remove a tag and then recreate it.
In this case it might have to do with https://github.com/pyocd/cmsis-pack-manager/blob/2d4c31c8d290c49ba5c00b164295d692efb1413b/.git_archival.txt and https://github.com/pyocd/cmsis-pack-manager/blob/2d4c31c8d290c49ba5c00b164295d692efb1413b/.gitattributes.
A setup like that appears to lead to rebuild of the auto-generated source tarball once a new commit is pushed to the default branch (which happened with https://github.com/pyocd/cmsis-pack-manager/commit/2d4c31c8d290c49ba5c00b164295d692efb1413b).

Long story short: if the `.git_archival.txt` and `.gitattributes` are not needed, can they please be removed?

**Comment by mbrossard** [comment]
This sounds like https://github.com/pypa/setuptools_scm/issues/806
